### PR TITLE
CoreCacheHandler: Don't set key as tag

### PR DESCRIPTION
**Pull request by blankse**
## Changes in this pull request  
At the moment the cache saves a additional tag with the key.
I think this makes no sense and increase the cache size.

### <samp>🤖 Generated by Copilot at 3da8c03</samp>

Remove redundant cache item tagging in `CoreCacheHandler.php`. This improves the cache performance and avoids potential bugs.

### <samp>🤖 Generated by Copilot at 3da8c03</samp>

> _`tag` method removed_
> _cache items no longer self-tagged_
> _a spring cleaning_

### <samp>🤖 Generated by Copilot at 3da8c03</samp>

* Remove redundant cache item tagging with own key ([link](https://github.com/pimcore/pimcore/pull/15948/files?diff=unified&w=0#diff-cc4f4d6b07b1520be4f1763eddc77036da47c488a52a366cd943ead918976ab4L546),                              F

### CODE REVIEW

File: lib/Cache/Core/CoreCacheHandler.php [hunk 1]
```diff
@@ -543,7 +543,6 @@ function ($currentValue) {
         $item->set($data);
         $item->expiresAfter($lifetime);
         $item->tag($tags);
-        $item->tag($key);
```
Comment (dvesh3) [inline lib/Cache/Core/CoreCacheHandler.php:504]: BC Break? If someone is relying on this tag?

Comment (blankse) [reply lib/Cache/Core/CoreCacheHandler.php:504]: Why should someone do this? It can simply loaded/removed without this tag...

Comment (blankse) [reply lib/Cache/Core/CoreCacheHandler.php:504]: Should we add a config to enable/disable this?

Comment (brusch) [reply lib/Cache/Core/CoreCacheHandler.php:504]: In my opinion this is not a BC-break, as it doesn't change the API nor the behavior when using the API. This topic is very low-level, not documented and not intended to be used directly.

**Comment by dvesh3** [comment]
@blankse maybe it is better to rebase it on 11.1 and add to upgrade notes. wdyt?

**Comment by brusch** [comment]
I agree, it's an improvement that should go into `11.x` and scheduled for `11.1.0` incl. upgrade notes.

**Comment by blankse** [comment]
@dvesh3 @brusch I rebased to 11.x and add a release note.

**Comment by dvesh3** [comment]
@blankse could you please resolve the conflicts?

**Comment by blankse** [comment]
@dvesh3 Done

**Comment by dvesh3** [comment]
@blankse thanks!

**Pull request merged**